x-pack/plugins/apm/public/components/app/infra_overview/infra_tabs/empty_prompt.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiEmptyPrompt, EuiPageTemplate } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+
+export function EmptyPrompt() {
+  return (
+    <EuiPageTemplate
+      pageContentProps={{
+        color: 'transparent',
+      }}
+      template="centeredBody"
+    >
+      <EuiEmptyPrompt
+        iconType="metricsApp"
+        color="subdued"
+        title={
+          <h2>
+            {i18n.translate('xpack.apm.infraTabs.noMetricsPromptTitle', {
+              defaultMessage: 'No infrastructure data found',
+            })}
+          </h2>
+        }
+        titleSize="m"
+        body={
+          <p>
+            {i18n.translate('xpack.apm.infraTabs.noMetricsPromptDescription', {
+              defaultMessage:
+                'Try adjusting your time range or check if you have any metrics data set up.',
```

Comment:
Can we check (in a follow-up PR) if there is infrastructure data?

I would also suggest saying "infrastructure data" instead of "metrics data"

Reply:
Hi @katrin-freihofner, we are showing this message when we don't have the infrastructure data that we need (container ids, pod names and host names), so we are already checking for that, if that's what you mean.
Regarding the 'metrics data' I will change it to 'metrics integration' maybe, it's what this message is referring too.
What do you think @boriskirov?

Reply:
Metrics integration makes sense to me as it's the same language used in the Metrics app. A link to the metrics page (like Caue suggests) or to "add a metrics integration" seems like it could be useful here.

Reply:
@MiriamAparicio sorry, my original message was confusing. I would like to let the user know if they should adjust the time range or if there has never been data and they need to set it up. (basically splitting this message into two chases).

Reply:
Hi @katrin-freihofner, I'm not sure if we can check from the apm if the metrics integration is set up in order to be more precise on the message, but I think that case will be handled by the infrastructure team and displayed in the tables that we render.
This specific case it's when we don't receive infrastructure data on the apm documents for that service, and I don't know if this can be set up or not, maybe the best solution will be just add a more generic message and point to the apm [documentation](https://www.elastic.co/guide/en/apm/guide/current/data-model-metrics.html)

Reply:
Ok, that's unfortunate as we'd like to give the user very precise information about what went wrong. Thanks for looking into it.

Reply:
@katrin-freihofner  https://github.com/elastic/kibana/issues/127742

Reply:
> This specific case it's when we don't receive infrastructure data on the apm documents for that service, and I don't know if this can be set up or not, maybe the best solution will be just add a more generic message and point to the apm [documentation](https://www.elastic.co/guide/en/apm/guide/current/data-model-metrics.html)

@MiriamAparicio I don't have all the context but: 
If this is based on APM data, I assume we are doing a time range scoped lookup, and then showing the empty state message if we don't find any data. 
To get what @katrin-freihofner is asking for, we could make an additional request but not scoped by a time range. This way we'll know if this user doesn't have data in the current time range, or doesn't have data at all.

Reply:
@sqren the additional request sounds good, but will it have impact in the performance?
I feel that it's already not great

Reply:
You can run it in the background and set `terminate_after` to 1, it should be very fast.